homeassistant/components/arwn/sensor.py

```diff
@@ -117,58 +117,23 @@ def async_sensor_event_received(msg):
 class ArwnSensor(SensorEntity):
     """Representation of an ARWN sensor."""
 
+    _attr_should_poll = False
+
     def __init__(self, topic, name, state_key, units, icon=None):
         """Initialize the sensor."""
         self.hass = None
         self.entity_id = _slug(name)
-        self._name = name
-        # This mqtt topic for the sensor which is its uid
-        self._uid = topic
         self._state_key = state_key
-        self.event = {}
-        self._unit_of_measurement = units
-        self._icon = icon
+        self._attr_name = name
+        # This mqtt topic for the sensor which is its uid
+        self._attr_unique_id = topic
+        self._attr_unit_of_measurement = units
+        self._attr_icon = icon
 
     def set_event(self, event):
         """Update the sensor with the most recent event."""
-        self.event = {}
-        self.event.update(event)
+        event = {}
+        event.update(event)
+        self._attr_state = event.get(self._state_key, None)
+        self._attr_extra_state_attributes = event
```

Comment:
```suggestion
        ev = {}
        ev.update(event)
        self._attr_state = ev.get(self._state_key, None)
        self._attr_extra_state_attributes = ev
```
if you set `event` to `{}` you override the data received by the `event` parameter

Reply:
Well spotted. Thanks